serde-support/src/main/java/io/micronaut/serde/support/serdes/DefaultFormattedTemporalSerde.java

```diff
@@ -125,5 +92,23 @@ public final T deserializeNonNull(Decoder decoder, DecoderContext decoderContext
      * @throws IOException if something goes wrong during deserialization
      */
     protected abstract T deserializeNonNullWithoutFormat(Decoder decoder, DecoderContext decoderContext, Argument<? super T> type) throws IOException;
-    
+
+    @NonNull
+    private DateTimeFormatter getFormatter(@NonNull SerdeConfiguration configuration) {
+        // Creates a custom formatter or returns the default one
+        return configuration.getDateFormat()
+            .map(format -> this.createFormatter(format, configuration))
+            .orElseGet(this::getDefaultFormatter);
+    }
+
+    @NonNull
+    private DateTimeFormatter createFormatter(@NonNull String pattern, @NonNull SerdeConfiguration configuration) {
```

Comment:
would not be easier to have this method be 

```@NonNull private Optional<DateTimeFormatter> createFormatter( @NonNull SerdeConfiguration configuration) {```

and replace 

```java
@NonNull
     private DateTimeFormatter getFormatter(@NonNull SerdeConfiguration configuration) {
         // Creates a custom formatter or returns the default one. 
         return configuration.getDateFormat()
             .map(format -> this.createFormatter(format, configuration))
             .orElseGet(this::getDefaultFormatter);
     }
```

with : 

```java
/**
 * Creates a custom formatter or returns the default one.
 */
@NonNull
     private DateTimeFormatter getFormatter(@NonNull SerdeConfiguration configuration) {
         return createFormatter(configuration)
             .orElseGet(this::getDefaultFormatter);
     }
```

Reply:
Thanks for your feedback. I've just pushed some changes to refactor these methods.